File: 20230529_Monday/20230529.go
// Leetcode: 1603. Design Parking System.
// https://leetcode.com/problems/design-parking-system/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 46 ms, faster than 33.33% of Go online submissions for Design
// Parking System.
// Memory Usage: 7.9 MB, less than 37.18% of Go online submissions for Design
// Parking System.
// 2023.05.29 Daily Challenge.
package main

type ParkingSystem struct {
	x [3]int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	ans := ParkingSystem{}
	ans.x[0] = big
	ans.x[1] = medium
	ans.x[2] = small
	return ans

}

func (this *ParkingSystem) AddCar(carType int) bool {
	t := carType - 1
	if this.x[t] <= 0 {
		return false
	} else {
		this.x[t] -= 1
		return true
	}

}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */

File: 20230529_Monday/20230529.rb
# Leetcode: 1603. Design Parking System.
# https://leetcode.com/problems/design-parking-system/
# = = = = = = = = = = = = = =
# Accepted.
# Thanks God, Jesus Christ!
# = = = = = = = = = = = = = =
# Runtime: 118 ms, faster than 76.00% of Ruby online submissions for Design
# Parking System.
# Memory Usage: 211.3 MB, less than 8.00% of Ruby online submissions for Design
# Parking System.
# 2023.05.29 Daily Challenge.
class ParkingSystem

=begin
    :type big: Integer
    :type medium: Integer
    :type small: Integer
=end
    def initialize(big, medium, small)
      @x = [big, medium, small]
        
    end


=begin
    :type car_type: Integer
    :rtype: Boolean
=end
    def add_car(car_type)
      t = car_type - 1
      if @x[t] <= 0
        return false
      else
        @x[t] -= 1
        return true
      end
    end


end

# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem.new(big, medium, small)
# param_1 = obj.add_car(car_type)
